cmd/loaders.go

```diff
@@ -16,38 +16,30 @@ import (
 
 const (
 	meshPrefix        = "MESH_"
-	maeshPrefix       = "MAESH_"
 	traefikMeshPrefix = "TRAEFIK_MESH_"
 )
 
 // EnvLoader loads a configuration from all the environment variables.
 type EnvLoader struct{}
 
-// Load loads the command's configuration from the environment variables prefixed with "TRAEFIK_MESH_" or "MAESH_".
-// The "MAESH_" prefix is deprecated and will be removed in the next major release.
-// If "TRAEFIK_MESH_" and "MAESH_" env variables are mixed up an error is returned.
+// Load loads the command's configuration from the environment variables prefixed with "TRAEFIK_MESH_".
 // As it is not possible to have a prefix with multiple "_" everything is normalized to "MESH_" under the hood for the decoding.
 func (e *EnvLoader) Load(_ []string, cmd *cli.Command) (bool, error) {
 	logger := logrus.StandardLogger()
+	traefikMeshVars := env.FindPrefixedEnvVars(os.Environ(), traefikMeshPrefix, cmd.Configuration)
 
-	traefikMeshVars := findAndNormalizePrefixedEnvVars(traefikMeshPrefix, cmd.Configuration)
-	maeshVars := findAndNormalizePrefixedEnvVars(maeshPrefix, cmd.Configuration)
+	var meshVars []string
 
-	if len(maeshVars) > 0 && len(traefikMeshVars) > 0 {
-		return false, fmt.Errorf("environment variable prefixed by %q cannot be mixed with variable prefixed by %q", maeshPrefix, traefikMeshPrefix)
+	for _, v := range traefikMeshVars {
+		meshVars = append(meshVars, strings.Replace(v, traefikMeshPrefix, meshPrefix, 1))
```

Comment:
Out of curiosity, why do we need to replace `traefikMeshPrefix` by `meshPrefix`? What would be the difference with just passing `traefikMeshPrefix` to `env.Decode`?

Reply:
Because it is currently not possible to have a prefix with multiple "_" in paerser.